### PR TITLE
[HND-00014] 🐪 feat: add hound.emitBatch() for single-pipeline batch emission

### PR DESCRIPTION
**Pull request by bllryy**
Emit multiple jobs in one Redis round trip instead of N. All state key                                                                                  
  SETs and queue ZADDs are batched into a single pipeline.                                                                                                
                                                                                                                                                          
  - Add EmitBatchEntry to public types                                                                                                                    
  - Add hound.emitBatch(jobs) to Hound class                                                                                                              
  - Update POST /emit/batch gateway to use hound.emitBatch()                                                                                              
  - Add 4 tests (order, empty input, all processed, deterministic ids)                                                                                    
  - Update emitting-jobs docs and README                                                                                                                  
  - Mark ROADMAP item done

### CODE REVIEW

File: core/libs/hound/mod.ts [hunk 1]
```diff
@@ -360,6 +361,31 @@ export class Hound<
...
+        pipe.set(stateKey, dataJson, 'EX', ttl);
+      } else {
+        pipe.set(stateKey, dataJson);
+      }
+    }
+    for (const [jobId, queue, , , score] of payloads) {
+      pipe.zadd(`queues:${queue}:q`, score, jobId);
+    }
+
+    await pipe.exec();
```
Comment (mirairoad) [inline core/libs/hound/mod.ts]: pipe.exec() doesn't reject on per-op errors returns [[err, val], ...]. Discarding the result + returning all jobIds means caller can hold a jobId for a job that didn't land. Wrap in MULTI/EXEC so it's all-or-nothing per chunk.

File: core/libs/hound/mod.ts [hunk 2]
```diff
@@ -360,6 +361,31 @@ export class Hound<
     return jobId;
   }
 
+  async emitBatch(jobs: EmitBatchEntry[]): Promise<string[]> {
```
Comment (mirairoad) [inline core/libs/hound/mod.ts]: we probably should cap the number of jobs into the batch to mirror consumer's claimCount. A 10k emit should fan out into 50 atomic transactions, not one unbounded pipeline. Redis is single-threaded; unbounded blocks every other client.

also emit/emitAsync/emitAndWait are bound onto ctx for in-handler use. emitBatch isn't. we can add easy in there too +)

File: core/libs/gateways/gateway.ts [hunk 3]
```diff
@@ -91,9 +91,7 @@ export function createGateway(options: GatewayOptions): Deno.HttpServer<Deno.Net
             options?: EmitOptions;
           }>;
           if (!Array.isArray(jobs)) return json({ error: 'body must be an array' }, 400);
```
Comment (mirairoad) [inline core/libs/gateways/gateway.ts:93]: Only validates Array.isArray(jobs). First entry without event throws inside #buildPayload and the whole batch dies with a generic 500. Validate each entry has a non-empty event, return 400 with offending index.

**Review by mirairoad — CHANGES_REQUESTED** [review]
